sanic/app.py

```diff
@@ -1552,10 +1552,19 @@ def _loop_add_task(
         name: Optional[str] = None,
         register: bool = True,
     ) -> Task:
-        prepped = cls._prep_task(task, app, loop)
-        task = loop.create_task(prepped, name=name)
+        if not isinstance(task, Future):
+            prepped = cls._prep_task(task, app, loop)
+            if sys.version_info == (3, 7):
```

Comment:
To keep the check consistent with `get_task`:
```suggestion
            if sys.version_info < (3, 8):
```